projects/core/modules/notifications/notification-alert/notification-alert.component.ts

```diff
@@ -20,16 +26,16 @@ export class TuiNotificationAlertComponent<O, I> {
     item?: NotificationAlert<O, I>;
 
     constructor(
-        @Inject(TuiDestroyService)
-        destroy$: TuiDestroyService,
+        @Inject(ElementRef) {nativeElement}: ElementRef<HTMLElement>,
+        @Inject(TuiDestroyService) destroy$: TuiDestroyService,
     ) {
         timer(ALERT_AUTOCLOSE_TIMEOUT)
-            .pipe(takeUntil(destroy$))
-            .subscribe(() => {
-                if (this.safeItem.autoClose) {
-                    this.closeDialog();
-                }
-            });
+            .pipe(
+                takeUntil(race(destroy$, fromEvent(nativeElement, 'mouseenter'))),
```

Comment:
You forgot to break that into two takeUntils. See my first comment ;)

Reply:
Done